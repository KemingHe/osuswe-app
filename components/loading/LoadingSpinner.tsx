import Image from 'next/image';
import type { JSX } from 'react';

import styles from '@/components/loading/LoadingSpinner.module.css';

export interface LoadingSpinnerProps {
  message?: string;
}

export function LoadingSpinner({
  message = 'Loading...',
}: LoadingSpinnerProps): JSX.Element {
  return (
    <output className="card text-center" aria-live="polite">
      <div className="card-body items-center gap-3 text-lg">
        <Image
          className={styles['loading-spinner-image']}
          src="/images/osuswe-logo-512.png"
          alt="Loading spinner"
          width={50}
          height={50}
          priority={true}
        />
        <p className="text-center">{message}</p>
      </div>
    </output>
  );
}
