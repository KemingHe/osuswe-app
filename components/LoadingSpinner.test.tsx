import { type RenderResult, render, screen } from '@testing-library/react';

import { LoadingSpinner } from '@/components/LoadingSpinner';

describe('LoadingSpinner component', () => {
  it('renders an accessible and visible loading spinner', () => {
    render(<LoadingSpinner />);

    // Select the element by its "status" role.
    const elementByStatus = screen.getByRole('status');
    expect(elementByStatus).toBeVisible();

    // Select the element by its text content.
    const elementByText = screen.getByText('Loading...');
    expect(elementByText).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment }: RenderResult = render(<LoadingSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
