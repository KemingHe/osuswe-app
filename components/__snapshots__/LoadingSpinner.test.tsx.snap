// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner component > matches the snapshot 1`] = `
<DocumentFragment>
  <output
    aria-live="polite"
    class="card bg-base-200 shadow-lg text-center"
  >
    <div
      class="card-body items-center gap-3 text-lg"
    >
      <img
        alt="Loading spinner"
        class="loading-spinner-image"
        data-nimg="1"
        decoding="async"
        height="50"
        src="/_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=128&q=75"
        srcset="/_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
      <p
        class="text-center"
      >
        Loading...
      </p>
    </div>
  </output>
</DocumentFragment>
`;
